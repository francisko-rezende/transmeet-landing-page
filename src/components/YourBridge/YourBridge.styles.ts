import styled, { css } from 'styled-components'

// import pic01 from '../../../public/images/pic-01.webp'

export const Wrapper = styled.section`
  display: grid;
  grid-template-columns: 5fr 7fr;
  margin-top: 11.8rem;

  @media (max-width: 811px) {
    grid-template-columns: 1fr;
    grid-template-rows: repeat(2, 397px);
  }
`

export const ContentWrapper = styled.div`
  @media (max-width: 811px) {
    grid-row: 2/3;
    margin-top: 3.2rem;
  }
`

export const Title = styled.h2`
  font-size: 7.2rem;
  font-weight: ${({ theme }) => theme.font.weight.bold};
  color: ${({ theme }) => theme.colors.typographyColors.black};

  @media (max-width: 811px) {
    text-align: center;
    font-size: 5.2rem;
  }
`

export const Paragraph = styled.p`
  margin-top: 5.1rem;
  text-align: left;
  line-height: 1.5;
  max-width: 46.9rem;

  @media (max-width: 811px) {
    text-align: center;
    font-size: 1.8rem;
    margin: 3.2rem auto;
  }

  ${({ theme }) => css`
    color: ${theme.colors.typographyColors.gray};
    font-weight: ${theme.font.weight.regular};
    font-size: ${theme.font.sizes['7']};
  `};
`

export const ImgWrapper = styled.div`
  position: relative;
  /* background-color: red; */

  @media (max-width: 811px) {
    grid-row: 1/2;
    display: flex;
    justify-content: center;
  }

  /* &::after {
    content: '';
    background-image: url('/images/pic-01.webp');
    background-size: cover;
    display: inline-block;
    position: absolute;
    width: 915px;
    height: 915px;
    left: 0;
    top: 0px;
    transform: translate(8%, -30%);
  } */
  /* position: absolute;
  top: 0;
  left: 50px; */
`
export const Img = styled.img`
  position: absolute;
  /* display: block;
  max-width: 100%;
  max-height: 100%; */
  width: 915px;
  height: 915px;
  pointer-events: none;
  transform: translateX(8%) translateY(-29%);

  @media (max-width: 811px) {
    transform: translateX(0%) translateY(-29%);
    width: 600px;
    height: 600px;
    /* transform: unset; */
    position: relative;
  }
`

export const AppStoreButtonsWrapper = styled.div`
  margin-top: 8.2rem;
  display: flex;
  justify-content: center;
  gap: 24px;

  @media (max-width: 811px) {
    margin-top: 4.1rem;
  }
`
