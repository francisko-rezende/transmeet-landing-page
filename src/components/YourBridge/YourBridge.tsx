import React from 'react'

import MainContainer from 'components/MainContainer'
import Image from 'next/image'

import * as S from './YourBridge.styles'

const YourBridge = () => {
  return (
    <MainContainer>
      <S.Wrapper>
        <S.ContentWrapper>
          <S.Title>
            Somos a sua ponte para a <span>transformação!</span>
          </S.Title>
          <S.Paragraph>
            Acreditamos que a educação é feita para qualquer pessoa, então
            criamos através do nosso app e plataforma, pontes para uma educação
            mais inclusiva.{' '}
          </S.Paragraph>
          <S.AppStoreButtonsWrapper>
            <Image
              src="/images/google.svg"
              width={176}
              height={60}
              alt="Em breve no Google Play"
            />
            <Image
              src="/images/apple.svg"
              width={176}
              height={60}
              alt="Em breve na App Store"
            />
          </S.AppStoreButtonsWrapper>
        </S.ContentWrapper>
        <S.ImgWrapper>
          <S.Img
            src="/images/pic-01-2.webp"
            // width={915}
            // height={915}
            alt="Um estudante sorrindo"
          />
          {/* <Image
            layout="fixed"
            objectFit="cover"
            src="/images/pic-01.webp"
            width={915}
            height={915}
            alt="Um estudante sorrindo"
          /> */}
        </S.ImgWrapper>
      </S.Wrapper>
    </MainContainer>
  )
}

export default YourBridge
