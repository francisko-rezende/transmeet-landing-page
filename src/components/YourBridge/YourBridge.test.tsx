import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import YourBridge from '.'

describe('<YourBridge />', () => {
  test('should render the section with all its elements', () => {
    const { container } = renderWithTheme(<YourBridge />)

    expect(
      screen.getByRole('heading', {
        name: /somos a sua ponte para a transformação!/i
      })
    ).toBeInTheDocument()
    expect(
      screen.getByText(
        /acreditamos que a educação é feita para qualquer pessoa, então criamos através do nosso app e plataforma, pontes para uma educação mais inclusiva\./i
      )
    ).toBeInTheDocument()
    expect(
      screen.getByRole('img', { name: /em breve no google play/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('img', { name: /em breve na app store/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('img', { name: /um estudante sorrindo/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
