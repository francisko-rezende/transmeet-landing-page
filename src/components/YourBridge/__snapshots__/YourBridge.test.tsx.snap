// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YourBridge /> should render the section with all its elements 1`] = `
.c0 {
  max-width: calc(1170px + 44px);
  padding: 0 22px;
  margin: 0 auto;
}

.c1 {
  display: grid;
  grid-template-columns: 5fr 7fr;
  margin-top: 11.8rem;
}

.c3 {
  font-size: 7.2rem;
  font-weight: 700;
  color: #152536;
}

.c4 {
  margin-top: 5.1rem;
  text-align: left;
  line-height: 1.5;
  max-width: 46.9rem;
  color: #68717A;
  font-weight: 400;
  font-size: 21px;
}

.c6 {
  position: relative;
}

.c7 {
  position: absolute;
  width: 915px;
  height: 915px;
  pointer-events: none;
  -webkit-transform: translateX(8%) translateY(-29%);
  -ms-transform: translateX(8%) translateY(-29%);
  transform: translateX(8%) translateY(-29%);
}

.c5 {
  margin-top: 8.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 24px;
}

@media (max-width:809px) {
  .c1 {
    grid-template-columns: 1fr;
    grid-template-rows: repeat(2,397px);
  }
}

@media (max-width:809px) {
  .c2 {
    grid-row: 2/3;
    margin-top: 3.2rem;
  }
}

@media (max-width:809px) {
  .c3 {
    text-align: center;
    font-size: 5.2rem;
  }
}

@media (max-width:809px) {
  .c4 {
    text-align: center;
    font-size: 1.8rem;
    margin: 3.2rem auto;
  }
}

@media (max-width:809px) {
  .c6 {
    grid-row: 1/2;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:809px) {
  .c7 {
    -webkit-transform: translateX(0%) translateY(-29%);
    -ms-transform: translateX(0%) translateY(-29%);
    transform: translateX(0%) translateY(-29%);
    width: 600px;
    height: 600px;
    position: relative;
  }
}

@media (max-width:809px) {
  .c5 {
    margin-top: 4.1rem;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Somos a sua ponte para a 
        <span>
          transformação!
        </span>
      </h2>
      <p
        class="c4"
      >
        Acreditamos que a educação é feita para qualquer pessoa, então criamos através do nosso app e plataforma, pontes para uma educação mais inclusiva.
         
      </p>
      <div
        class="c5"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27176%27%20height=%2760%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Em breve no Google Play"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27176%27%20height=%2760%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Em breve na App Store"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
    <div
      class="c6"
    >
      <img
        alt="Um estudante sorrindo"
        class="c7"
        src="/images/pic-01-2.webp"
      />
    </div>
  </section>
</div>
`
